pkg/kubelet/dockertools/docker.go

```diff
@@ -1049,3 +1049,21 @@ func RunContainer(client DockerInterface, container *api.Container, pod *api.Pod
 	}
 	return dockerContainer.ID, nil
 }
+
+// KillContainer kills a docker container with the given containerID.
+// TODO(yifan): To use strong type for the containerID.
+func KillContainer(client DockerInterface, containerID string,
```

Comment:
I liked your suggestion to have the runtime do KillContainer() and handle KillPod() in the Kubelet for now. This will let is set readiness and record events. Is this what you had in mind?